### ISSUE TITLE
Runtime error with Angular 12

### ISSUE THREAD

**Issue by yschroeder** [sclausen/ngx-mqtt#184]
- [x] **I have searched for similar issues in this repository, but couldn't find one.**
- [x] **I have read the README and have a basic understanding how angular works.**
## I'm submitting a...
- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [x] Bug report  
- [ ] Feature request

## Current behavior
`ngx-mqtt` does not work with Angular 12.  It creates a runtime error in the browser console:
```
util.js:103 Uncaught ReferenceError: Buffer is not defined
```
I am using `ngx-mqtt` version 8.0.1, as I am unable to install 8.0.3 (see #169).

## Expected behavior
`ngx-mqtt` should work as it does with Angular 11.

## [Minimal working example](https://en.wikipedia.org/wiki/Minimal_working_example) of the problem with instructions
https://github.com/yschroeder/angular12-ngx-mqtt

Instructions:
*  Run `ng serve`
* Open browser
* Page stays blank
* Console shows error

## What is the motivation / use case for changing the behavior?
I'd like to upgrade my project to Angular 12, but `ngx-mqtt` stops me from doing it.

## Environment

<pre><code>
ngx-mqtt version: 8.0.1, cannot use 8.0.3 due to #169

- Broker Mosquitto 2.0.10
</code></pre>

## Further information

I looked into this myself already a bit and found that Angular 12 uses Webpack 5. Webpack 5 does not provide polyfills for Node.js functionality anymore. So while the `postinstall` hook tries to enable the Node.js polyfills, this does nothing, as Webpack 5 still does not provide them.

When the bundling of `mqtt.min.js` which is discussed in #169 is actually implemented, this problem will likely be resolved as well, as the dependency on Node.js polyfill will be gone.

All in all, this issue is more of a documentation, so others can figure out why their project does not work with Angular 12. Further, it increases the urgency of solving #169.

**Comment by sclausen** [comment]
This should be resolved now. Can you confirm installing works with ngx-mqtt@9.0.1?

**Comment by yschroeder** [comment]
I can confirm that ngx-mqtt@9.0.1 works with Angular 12.0.3.

However, I had 12.0.2 installed before and you have 12.0.3 as dependency, so installation failed on Angular 12.0.2.

**Comment by sclausen** [comment]
@yschroeder It actually fails, or just complains?

**Comment by yschroeder** [comment]
It fails:

```
$ npm install ngx-mqtt@9.0.1 --save
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: ngx-mqtt@8.0.1
npm ERR! node_modules/ngx-mqtt
npm ERR!   ngx-mqtt@"9.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! ngx-mqtt@"9.0.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @angular/common@12.0.3
npm ERR! node_modules/@angular/common
npm ERR!   peer @angular/common@"^12.0.2" from ngx-mqtt@9.0.1
npm ERR!   node_modules/ngx-mqtt
npm ERR!     ngx-mqtt@"9.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/yannic/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/yannic/.npm/_logs/2021-06-06T11_29_52_150Z-debug.log
```

This happens on my project that was updated to Angular 12 a few weeks ago. If I use the old Angular 11 version and freshly update it to Angular 12 it works.

The old version is not 12.0.2 but actually 12.0.0. I just checked again.

TL;DR:
Updating ngx-mqtt from 8.0.1 to 9.0.1 on Angular 12.0.0 fails.
Updating ngx-mqtt from 8.0.1 to 9.0.1 on Angular 12.0.3 works.

**Comment by sclausen** [comment]
Thank you very much, I’ll try to use proper version ranges. Give me a couple of hours.

**Comment by sclausen** [comment]
@yschroeder There are several issues with dependencies and currently even `ng new project-name` doesn't work but I managed to build a new version which at least works with angular 12 (for me).
Can you confirm that it works for you?

**Comment by yschroeder** [comment]
I can confirm 9.0.4 works with Angular 12.0.0

**Comment by sclausen** [comment]
Thank you very much!
